tiledb/sm/filesystem/CMakeLists.txt

```diff
@@ -39,12 +39,39 @@ commence(object_library vfs)
         vfs.cc
         vfs_file_handle.cc
         win.cc
+        ../curl/curl_init.cc
     )
     if (TILEDB_S3)
+        target_compile_definitions(vfs INTERFACE -DHAVE_S3)
+        target_include_directories(vfs
+                PRIVATE
+                "${CMAKE_CURRENT_SOURCE_DIR}/../../../external/include"
+        )
+        add_subdirectory(s3)
         list(APPEND VFS_SOURCES
             s3.cc
             s3_thread_pool_executor.cc
         )
+        if(WIN32)
```

Comment:
So it turns out because we never had the `target_compile_definitions(vfs INTERFACE -DHAVE_S3)`. The object library was never building the s3 code. We never noticed until now when I was wanting to add the storage provider. I can remove this from this PR but we'll need to circle back to enable this for the object library.